src/components/Brewery/Brewery.js

```diff
@@ -28,18 +52,50 @@ const MapWrapper = styled.section`
 
 const Brewery = (props) => {
 
+  const handleFavorites = () => {
+    if (props.favorites.length > 0) {
+      props.favorites.forEach(favorite => {
+        if (favorite.id == props.pub.id) {
+          props.setFavorites(props.pub);
+        } else {
+          props.setFavorites(props.pub);
+        }
+      })
+    } else {
+      props.setFavorites(props.pub);
+    }
+  }
+
+  const checkFavorites = () => {
+    if (props.favorites.length > 0) {
+      props.favorites.forEach(favorite => {
+        if (favorite.id == props.pub.id) {
+          return false;
+        } else {
+          return true;
+        }
+      })
+    } else {
+      return true;
+    }
+  }
```

Comment:
Called on render, checks to see if it is in favorites and renders the save button styling accordingly, saying either "remove from favorites" or "favorite this brewery"